src/chainstate/stacks/db/blocks.rs

```diff
@@ -5143,6 +5147,45 @@ impl StacksChainState {
         let evaluated_epoch = clarity_tx.get_epoch();
         clarity_tx.reset_cost(parent_block_cost.clone());
 
+        debug!("Evaluating block with epoch = {}", evaluated_epoch);
```

Comment:
Can you lift this block into its own function?  This function is already pretty long.

Reply:
Sure

Reply:
Thanks!